agent/agent_test.go

```diff
@@ -1305,90 +1335,89 @@ task_groups:
 	}, []string{panicLog})
 }
 
-func (s *AgentSuite) TestTeardownTaskSucceeds() {
-	s.tc.taskGroup = "task_group_name"
+func (s *AgentSuite) TestTeardownTaskFails() {
+	const taskGroup = "task_group_name"
 	projYml := `
 task_groups:
   - name: task_group_name
+    teardown_task_can_fail_task: true
     teardown_task:
       - command: shell.exec
         params:
-          script: exit 0
+          script: exit 1
 `
 	s.setupRunTask(projYml)
-	s.tc.taskConfig.Task.TaskGroup = s.tc.taskGroup
+	s.tc.taskConfig.Task.TaskGroup = taskGroup
+	s.tc.taskConfig.TaskGroup = *s.tc.taskConfig.Project.FindTaskGroup(taskGroup)
 
-	s.NoError(s.a.runPostTaskCommands(s.ctx, s.tc))
+	s.Error(s.a.runPostTaskCommands(s.ctx, s.tc))
 
 	s.NoError(s.tc.logger.Close())
 	checkMockLogs(s.T(), s.mockCommunicator, s.tc.taskConfig.Task.Id, []string{
 		"Running post-task commands",
```

Comment:
What happened to the other `checkMockLogs` checks in this test?

Reply:
got messed up in the the merge, I reset it.